### PR TITLE
Expose more of istioctl as library

### PR DESCRIPTION
**Pull request by nmittler**
This is another step toward removing the istioctl test component.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -286,18 +292,28 @@ func newClientInternal(clientFactory util.Factory, revision string) (*client, er
 	c.extSet = ext
 	c.versionClient = ext
 
+	// We are running a one-off command locally, so we don't need to worry too much about rate limitting
+	// Bumping this up greatly decreases install time
+	restConfig := c.RESTConfig()
+	restConfig.QPS = 50
```
Comment (howardjohn) [inline pkg/kube/client.go:298]: this is not universal - can we pass this in like we do for pilot?
```
		s.kubeRestConfig, err = kubelib.DefaultRestConfig(args.RegistryOptions.KubeConfig, "", func(config *rest.Config) {
			config.QPS = 20
			config.Burst = 40
		})
		if err != nil {
			return fmt.Errorf("failed creating kube config: %v", err)
		}

		s.kubeClient, err = kubelib.NewClient(kubelib.NewClientConfigForRestConfig(s.kubeRestConfig))
		if err != nil {
			return fmt.Errorf("failed creating kube client: %v", err)
		}
```

**Review by howardjohn — COMMENTED** [review]
lgtm aside from qps comment

File: operator/cmd/mesh/manifest-apply.go [hunk 2]
```diff
@@ -144,8 +146,12 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, maArgs *manifestApplyAr
 	if err := configLogs(logOpts); err != nil {
 		return fmt.Errorf("could not configure logs: %s", err)
 	}
-	if err := ApplyManifests(applyFlagAliases(maArgs.set, maArgs.manifestsPath, maArgs.revision), maArgs.inFilenames, maArgs.force, rootArgs.dryRun,
-		maArgs.kubeConfigPath, maArgs.context, maArgs.readinessTimeout, l); err != nil {
+	client, err := kube.NewClient(kube.BuildClientCmd(maArgs.kubeConfigPath, maArgs.context))
+	if err != nil {
```
Comment (ayj) [inline operator/cmd/mesh/manifest-apply.go:150]: `K8sConfig` creates a client with increased QPS/Burst to reduce install time. This behavior should probably be preserved.

Comment (howardjohn) [reply operator/cmd/mesh/manifest-apply.go:150]: +1, LGTM otherwise

Comment (nmittler) [reply operator/cmd/mesh/manifest-apply.go:150]: Added the configuration to kube client creation.

**Comment by nmittler** [comment]
@ayj @howardjohn PTAL

**Review by ayj — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**